endpoints/users.js

```diff
@@ -5,24 +5,40 @@ const config = require('../config');
 
 create = (request, response) => {
     if (!request.body.username) {
-        response.status(400).json({"err": "Missing username field."});
-    }
-    if (!request.body.password) {
-        response.status(400).json({"err": "Missing password field."});
-    }
+        response.status(400).json({"err": "Missing username field."}).end();
```

Comment:
I suggest two approaches to making this code cleaner.

The first would be to use this NPM library that I found called `express-validator`. This would make validating the fields easier, as you would just have to add in middleware functions in order to do so. We could also create our own custom validators to do things such as checking for an already existing user. This may seem excessive for such a small feature, however we will eventually have to validate more fields. For instance, we will probably include email one day so that password reset and email confirmation can exist. On top of this, it allows for your JSON error output to be very nice and organized with little code. In other words, don't reinvent the wheel.

The second inferior approach if you don't wish to install a library for some reason is to just write middleware functions in a seperate file, like `validators.js`, and these would add errors to an array at `request.validationErrors`, which we could check for in the endpoint. 

With the route map approach, I think to add endpoint specific middleware we would simply do something like this:
```
{
    "/my_endpoint": {
        get: [middlewareOne, middlewareTwo, endpointFunct]
    }
}
```

I believe express works in this way where you can pass an array of functions to `app.get` or whatever and it will execute it as a middleware stack. If that isn't the case, I might consider abandoning the route-map approach for using Express routers instead.